R/plotLoadings.R

```diff
@@ -0,0 +1,374 @@
+        
+        if (layout == "tree") {
+            # Plot tree with feature loadings
+            p <- .loadings_tree_plotter(x, loadings_matrix, ncomponents, tree_name)
+        } else {
+            # Plot features with the layout selected
+            p <- .plot_pca_feature_loadings(L, layout, ncomponents)
+        }
+    return(p)
+    }
+)
+#' @rdname plotLoadings
+#' @export 
+setMethod("plotLoadings", signature = c(x = "matrix"),
+    function(x,
+            dimred = "PCA",
+            layout = "heatmap",
+            n = 10,
+            ncomponents = 5,
+            tree_name = "phylo",
+            ...) {
+        # Making sure there is no error in parameters given by the user
+        .check_parameters(x,
+                        dimred = dimred,
+                        layout = layout,
+                        n = n,
+                        ncomponents = ncomponents,
+                        tree_name = tree_name,
+                        ...)
+                      
+        #Checking if there are enough components in the matrix
+        .check_components(x, ncomponents)
+        # Ordering loadings and adding factor to keep the order
+        df <- .get_loadings_plot_data(x, n, ncomponents)
+        # Plot features with the layout selected
+        p <- .plot_pca_feature_loadings(df, layout, ncomponents)
+        
+        return(p)
+    }
+)
+
+.check_parameters <- function(x, dimred, layout, n, ncomponents, tree_name,...) {
+    # Check tree_name
+    if( is(x, "TreeSummarizedExperiment") && !(tree_name %in% rowTreeNames(x)  && .is_a_string(tree_name))){
+        stop("'tree_name' must be a single character value specifying a colTree.", call. = FALSE)
+    }
+    # Checking if dimred is correct
+    if( is(x, "TreeSummarizedExperiment") && !(dimred %in% reducedDimNames(x)  && .is_a_string(dimred))){
+        stop("'dimred' must specify reducedDim.", call. = FALSE)
+    }
+    # Checking if layout is correct
+    if ( !(layout %in% c("screeplot", "barplot", "tree", "heatmap") && .is_a_string(layout)) ) {
+        stop("'dimred' must be one of c('screeplot', 'barplot', 'tree', 'heatmap').", call. = FALSE)
+    }
+    # Making sure the user doesn't try to plot the tree if he gives only the matrix
+    if (is.matrix(x) && layout == "tree") {
+        stop("TreeSummarizedExperiment object is required for the tree plotting.", call. = FALSE)
+    }
+    # Making sure the tree is not null
+    if( layout == "tree" && !(is(x, "TreeSummarizedExperiment") && !is.null(rowTree(x, tree_name)) )) {
+        stop ("Tree is null.", call. = FALSE)
+    }
+    #Checking if n is a positive number
+    if ( !(is.numeric(n) && n > 0) ) {
+        stop("'n' must be a positive number.", call. = FALSE)
+    }
+    #Checking if ncomponents is a positive number 
+    if ( !(is.numeric(ncomponents) && ncomponents > 0) ) {
+        stop("'ncomponents' must be a positive number.", call. = FALSE)
+    }
+
```

Comment:
R functions always return something. I think it is good practice to specify the returned value even though the value is not used. it could be just return(NULL)